src/AspNet.Security.OAuth.Deezer/DeezerAuthenticationOptions.cs

```diff
@@ -0,0 +1,85 @@
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using System.Security.Claims;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.OAuth;
+using Microsoft.AspNetCore.Http;
+using static AspNet.Security.OAuth.Deezer.DeezerAuthenticationConstants;
+
+namespace AspNet.Security.OAuth.Deezer
+{
+    /// <summary>
+    /// Defines a set of options used by <see cref="DeezerAuthenticationHandler"/>.
+    /// </summary>
+    public class DeezerAuthenticationOptions : OAuthOptions
+    {
+        /// <summary>
+        /// Deezer API Permissions
+        /// <para>https://developers.deezer.com/api/permissions</para>
+        /// </summary>
+
+        public DeezerAuthenticationOptions()
+        {
+            ClaimsIssuer = DeezerAuthenticationDefaults.Issuer;
+
+            CallbackPath = new PathString(DeezerAuthenticationDefaults.CallbackPath);
+
+            AuthorizationEndpoint = DeezerAuthenticationDefaults.AuthorizationEndpoint;
+            TokenEndpoint = DeezerAuthenticationDefaults.TokenEndpoint;
+            UserInformationEndpoint = DeezerAuthenticationDefaults.UserInformationEndpoint;
+
+            ClaimActions.MapJsonKey(ClaimTypes.NameIdentifier, "id");
+            ClaimActions.MapJsonKey(ClaimTypes.Name, "name");
+            ClaimActions.MapJsonKey(ClaimTypes.GivenName, "firstname");
+            ClaimActions.MapJsonKey(ClaimTypes.Surname, "lastname");
+            ClaimActions.MapJsonKey(ClaimTypes.Email, "email");
+            ClaimActions.MapJsonKey(ClaimTypes.DateOfBirth, "birthday");
+            ClaimActions.MapJsonKey(ClaimTypes.Gender, "gender");
+            ClaimActions.MapJsonKey(ClaimTypes.Country, "country");
+
+            ClaimActions.MapJsonKey(Claims.Username, "name");
+            ClaimActions.MapJsonKey(Claims.Avatar, "picture");
+            ClaimActions.MapJsonKey(Claims.Avatar_XL, "picture_xl");
+            ClaimActions.MapJsonKey(Claims.Avatar_Big, "picture_big");
+            ClaimActions.MapJsonKey(Claims.Avatar_Medium, "picture_medium");
+            ClaimActions.MapJsonKey(Claims.Avatar_Small, "picture_small");
+            ClaimActions.MapJsonKey(Claims.Url, "link");
+            ClaimActions.MapJsonKey(Claims.Status, "status");
+            ClaimActions.MapJsonKey(Claims.Inscription_Date, "inscription_date");
+            ClaimActions.MapJsonKey(Claims.Language, "lang");
+            ClaimActions.MapJsonKey(Claims.IsKid, "is_kid");
+            ClaimActions.MapJsonKey(Claims.Tracklist, "tracklist");
+            ClaimActions.MapJsonKey(Claims.Type, "type");
+            ClaimActions.MapJsonKey(Claims.Explicit_Content_Level, "explicit_content_level");
+
+            Scope.Add(Scopes.Identity);
+        }
+
+        //
+        // Summary:
+        //     Gets or sets the provider-assigned app id.
+        public string App_Id
```

Comment:
These overrides are not necessary, users can just use `ClientId` and `ClientSecret`.

Reply:
Yes, but it's easier to understand, if someone is not fit in the OAuth enough.

Reply:
The providers are all based upon the same base class, and we'd rather than create lots of inconsistency and extra code to maintain to match the arbitrary names specific providers use for common concepts.

Reply:
Ok, I see. One standard is better.

Reply:
> The providers are all based upon the same base class, and we'd rather than create lots of inconsistency and extra code to maintain to match the arbitrary names specific providers use for common concepts.

I should mention. My Deezer Provider was completelly made from the Discord provider with some other small code blocks of other providers. I have written not much code myself. This means, many problems you have requested to solve are actually present one to one somewhere in another provider. Mostly of couse in Discord. 😄